### PR TITLE
add support for auto-converting to expected value type for easyconfig parameters

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86, @Caylo: please review?

File: test/framework/type_checking.py [hunk 1]
```diff
@@ -40,16 +41,46 @@ def test_check_type_of_param_value(self):
...
+        self.assertEqual(convert_value_type(100, basestring), '100')
+        self.assertEqual(convert_value_type((100,), str), '(100,)')
+        self.assertEqual(convert_value_type([100], basestring), '[100]')
+        self.assertEqual(convert_value_type(None, str), 'None')
+
+        # to int/float
+        self.assertEqual(convert_value_type('100', int), 100)
+        self.assertEqual(convert_value_type('0', int), 0)
+        self.assertEqual(convert_value_type('-123', int), -123)
+        self.assertEqual(convert_value_type('1.6', float), 1.6)
```
Comment (wpoely86) [inline test/framework/type_checking.py]: ` self.assertEqual(convert_value_type('1.6', int), 1)` ?

Comment (wpoely86) [inline test/framework/type_checking.py]: `self.assertEqual(convert_value_type('5', float), 5.0)` ?

**Comment by wpoely86** [comment]
lgtm

**Comment by Caylo** [comment]
nothing to add here :)

**Comment by boegel** [comment]
Thanks for the review @wpoely86 and @Caylo!

**Pull request merged**